### PR TITLE
JUnit tests, calc impl

### PR DESCRIPTION
**Pull request by DDemoNZ**
add implementation of calculator
add JUnit tests for base operation

### CODE REVIEW

File: src/main/java/core/basesyntax/Calc.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package core.basesyntax;
+
+import java.util.function.DoubleBinaryOperator;
+
+public class Calc {
```
Comment (Sofasmile) [inline src/main/java/core/basesyntax/Calc.java]: let's use full name of the class

File: src/test/java/core/basesyntax/CalcTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    private static final double DELTA = 0.0001;
+    private Calc calc;
+
+    @Before
+    public void setUp() throws Exception {
+        calc = new Calc();
+    }
+
+    @After
+    public void tearDown() throws Exception {
```
Comment (Sofasmile) [inline src/test/java/core/basesyntax/CalcTest.java]: do we need this method?

File: src/test/java/core/basesyntax/CalcTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Test
+    public void mulPositive() {
+        Assert.assertEquals(25, calc.calculate(5, 5, '*'), DELTA);
+    }
+
+    @Test
+    public void mulByZero() {
+        Assert.assertEquals(0, calc.calculate(5, 0, '*'), DELTA);
```
Comment (Sofasmile) [inline src/test/java/core/basesyntax/CalcTest.java]: let's add test with negative parameter

File: src/test/java/core/basesyntax/CalcTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    }
+
+    @Test
+    public void divPositive() {
+        Assert.assertEquals(5, calc.calculate(25, 5, '/'), DELTA);
```
Comment (Sofasmile) [inline src/test/java/core/basesyntax/CalcTest.java]: the same about negative parameter

File: src/test/java/core/basesyntax/CalcTest.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+        Assert.assertEquals(125, calc.calculate(5, 3, '^'), DELTA);
+    }
+
+    @Test
+    public void powerWithNegativeNum() {
+        Assert.assertEquals(0.04, calc.calculate(5, -2, '^'), DELTA);
+    }
+
+    @Test
+    public void powBy0() {
```
Comment (Sofasmile) [inline src/test/java/core/basesyntax/CalcTest.java]:
```suggestion
    public void powByZero() {
```
